File: Java/src/Implement strStr.java
public class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0 || needle.equals(haystack)){
            return 0;
        }
        if (needle.length() > haystack.length()){
            return -1;
        }
        
        for (int i=0; i <= (haystack.length() - needle.length()) ;i++){
            int n = 0;
            while(n < needle.length() && haystack.charAt(i+n) == needle.charAt(n))
                n++;
            if(n == needle.length())
                return i;
        }
        return -1;
         
    }
}

File: Python/Implement strStr.py
class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        if len(needle) == 0 or haystack == needle:
            return 0
        if len(needle) > len(haystack):
            return -1
        for i in range(len(haystack)-len(needle)+1):
            if haystack[i:i+len(needle)] == needle:
                return i
        return -1
        